### PR TITLE
Rename the default system namespace to kube-federation-system

### LINKED ISSUE #868: The default system namespace should be prefixed with `kube-`
As per the kube api convention [on system namespaces](https://github.com/kubernetes/community/blob/20c1e33a741f238ec0428f7fdc096ca1ff0d862d/contributors/devel/sig-architecture/api-conventions.md#namespace-names), the default federation system namespace should have the `kube-` prefix reserved for system namespaces to avoid collision with user namespaces.  Suggested options:

 - `kube-fed`
 - `kube-fed-system`
 - `kube-federation`
 - `kube-federation-system`

### PR DESCRIPTION
**Pull request by marun**
Fixes #868

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # Unreleased
+-  [#868](https://github.com/kubernetes-sigs/federation-v2/issues/868)
+   The default kubefed system namespace has been changed from
+   `federation-system` to `kube-federation-system`.  The `kube-`
+   prefix is reserved for system namespaces and including it avoids
+   having the the kubefed namespace conflict with a user namespace.
```
Comment (font) [inline CHANGELOG.md]: s/the //

Comment (marun) [reply CHANGELOG.md]: Done

**Comment by marun** [comment]
Updated

**Pull request merged**